exec/java-exec/src/main/java/org/apache/drill/exec/ops/OperatorContextImpl.java

```diff
@@ -130,6 +134,60 @@ public OperatorStats getStats() {
     return stats;
   }
 
+
```

Comment:
Oops.. I completely forgot about cancellation. ExecutorService makes the code cleaner too. I changed the method to return a ListenableFuture.